java-checks/src/main/java/org/sonar/java/checks/regex/RegexComplexityCheck.java

```diff
@@ -177,106 +165,6 @@ private boolean isCommented(LiteralTree regexPart) {
 
   }
 
-  private class ComplexityCalculator extends RegexBaseVisitor {
```

Comment:
I was initially confused because the code in the Java analyzer side is way more complex than the implementation on Commons. It turns out that the implementation on Commons contains only a subpart of the one from the analyzer, the analyzer having additional logic to count the complexity of each part individually if the regex is made out of multiple parts.
The implementation of this "ComplexityCalculator" is the same, so the migration is trivial.